Python/pywarpx/Diagnostics.py

```diff
@@ -6,6 +6,20 @@
 
 from .Bucket import Bucket
 
-diagnostics = Bucket('diagnostics', diags_names=[])
-diagnostics_list = []
+diagnostics = Bucket('diagnostics', _diagnostics_dict={})
+
+class Diagnostic(Bucket):
+    """
+    This is the same as a Bucket, but checks that any attributes are always given the same value.
+    """
+    def add_new_attr_with_check(self, name, value):
+        if name.startswith('_'):
+            self._localsetattr(name, value)
+        else:
+            if name in self.argvattrs:
+                assert value == self.argvattrs[name], Exception(f'Diagnostic attributes not consistent for {self.instancename}')
```

Comment:
I don't know if `f` is a variable, but I thought the `assert` in picmi using `Exception` doesn't need the `f`.
Only the string.
```suggestion
                assert value == self.argvattrs[name], Exception('Diagnostic attributes not consistent for {self.instancename}')
```

Reply:
Hi @LDAmorim The f-string is a new, very nice, way in Python of formatting strings. Here is the documentation https://docs.python.org/3/reference/lexical_analysis.html#index-24

Reply:
Thanks @dpgrote ! This is really nice indeed 👍